src/Magento/FunctionalTestingFramework/Allure/AllureHelper.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\FunctionalTestingFramework\Allure;
+
+use Yandex\Allure\Adapter\Allure;
+use Yandex\Allure\Adapter\Event\AddAttachmentEvent;
+
+class AllureHelper
+{
+    /**
+     * Adds attachment to the current step
+     * @param mixed  $data
+     * @param string $caption
+     * @throws \Yandex\Allure\Adapter\AllureException
+     * @return void
+     */
+    public static function addAttachmentToCurrentStep($data, $caption)
+    {
+        Allure::lifecycle()->fire(new AddAttachmentEvent($data, $caption));
+    }
+
+    /**
+     * Adds Attachment to the last executed step. Required due to Allure root-step behavior
+     * @param mixed  $data
+     * @param string $caption
+     * @return void
+     */
+    public static function addAttachmentToLastStep($data, $caption)
```

Comment:
Just curious about in general when to use this over the current step? I see you use this one in the curl handler, but I cant figure out why. Maybe add a bit to the phpdoc like:

`This function should be used when ___blahblahblah___`

Reply:
I'll add a comment to clear it up, the gist of it is that during a step (aka a function called in the test via `$I->doathing()`) you use current. If the framework is doign its own logic outisde a function like that, you can add artifact to the last step (which is usually a comment step)